pkg/library/libraryinputresources/resource.go

```diff
@@ -74,37 +101,22 @@ func LenientResourcesFromDirRecursive(location string) ([]*Resource, error) {
 	return currResourceList, errors.Join(errs...)
 }
 
-func findGVR(resources map[schema.GroupVersion]*metav1.APIResourceList, gvk schema.GroupVersionKind) (*schema.GroupVersionResource, error) {
-	apiResourceList, ok := resources[gvk.GroupVersion()]
-	if !ok {
-		return nil, fmt.Errorf("failed to find api resource list for gvk %s", gvk)
-	}
-
-	var matches []*schema.GroupVersionResource
-	for _, apiResource := range apiResourceList.APIResources {
-		if strings.Contains(apiResource.Name, "/") {
-			// Skip subresources
-			continue
-		}
-		if apiResource.Kind == gvk.Kind {
-			matches = append(matches, &schema.GroupVersionResource{
-				Group:    gvk.Group,
-				Version:  gvk.Version,
-				Resource: apiResource.Name,
-			})
-		}
+func findGVR(gvkToResources map[schema.GroupVersionKind][]schema.GroupVersionResource, gvk schema.GroupVersionKind) (*schema.GroupVersionResource, error) {
+	if gvkToResources == nil {
```

Comment:
nit: unnecessary.  You can read from a nil map

Reply:
Removed